js/fetch.ts

```diff
@@ -97,59 +99,127 @@ export const Headers = DomIterableMixin<string, string, typeof HeadersBase>(
   headerMap
 );
 
-class FetchResponse implements domTypes.Response {
+class Body implements domTypes.Body, domTypes.ReadableStream, io.Reader {
+  bodyUsed = false;
+  readonly body: domTypes.ReadableStream | null = null; // TODO
+  private _buffering = false;
+  private _bodyWaiter: Resolvable<ArrayBuffer>;
+  private _chunks: Uint8Array[] = [];
+  private _data: Uint8Array | null = null;
+  readonly locked: boolean = false; // TODO
+
+  constructor(readonly rid: number, readonly contentType: string) {
+    this._bodyWaiter = createResolvable();
+  }
+
+  private async _bufferBody(): Promise<void> {
+    if (!this._buffering) {
+      this._buffering = true;
+      try {
+        while (true) {
+          const b = new Uint8Array(64 * 1024);
```

Comment:
Is there no content-length header so we can just preallocate a buffer with the right size?

Reply:
This whole bit is preliminary and will be replaced with a general purpose method of buffering any readable object. 
See https://golang.org/pkg/bytes/#Buffer.ReadFrom and WIP https://github.com/ry/deno/blob/buffer/js/buffer.ts
I've added a TODO comment that discusses this.